src/live_restore/live_restore_fs.c

```diff
@@ -1395,90 +1389,240 @@ __live_restore_setup_lr_fh_directory(WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_
 }
 
 /*
- * __live_restore_setup_lr_fh_file --
- *     Populate a live restore file handle for a normal (non-directory) file.
+ * __live_restore_remove_temporary_file --
+ *     Remove a temporary file if it exists. If it does exist, log a warning.
  */
 static int
-__live_restore_setup_lr_fh_file(WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_FS *lr_fs,
-  const char *name, uint32_t flags, WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh)
+__live_restore_remove_temporary_file(
+  WT_SESSION_IMPL *session, WT_FILE_SYSTEM *os_fs, char *dest_path, char **tmp_file_path)
 {
-    bool dest_exist = false, source_exist = false, have_stop = false;
+    size_t tmp_file_path_len = strlen(dest_path) + strlen(".lr_tmp") + 1;
+    WT_RET(__wt_calloc(session, 1, tmp_file_path_len, tmp_file_path));
+    WT_RET(__wt_snprintf(*tmp_file_path, tmp_file_path_len, "%s.lr_tmp", dest_path));
+    /* Delete any existing temporary file. Also report a warning if it existed already. */
+    bool exists = false;
+    WT_RET(os_fs->fs_exist(os_fs, (WT_SESSION *)session, *tmp_file_path, &exists));
+    if (!exists)
+        return (0);
+    __wt_verbose_info(session, WT_VERB_LIVE_RESTORE,
+      "Found existing temporary file: %s deleting it!", *tmp_file_path);
+    return (os_fs->fs_remove(os_fs, (WT_SESSION *)session, *tmp_file_path, 0));
+}
+
+/*
+ * __live_restore_fs_atomic_copy_file --
+ *     Copy a file across for live restore. We intentionally do not call the WiredTiger copy and
+ *     sync function as we are copying between layers and that function copies between two paths.
+ *     This is the same "path" from the perspective of a function higher in the stack.
+ */
+static int
+__live_restore_fs_atomic_copy_file(WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_FS *lr_fs,
+  WT_FS_OPEN_FILE_TYPE type, const char *filename)
+{
+    WT_DECL_RET;
     WT_SESSION *wt_session = (WT_SESSION *)session;
+    char *buf = NULL, *source_path = NULL, *dest_path = NULL, *tmp_dest_path = NULL;
+    bool dest_closed = false;
+    size_t read_size = lr_fs->read_size;
 
-    WT_RET_NOTFOUND_OK(
-      __live_restore_fs_has_file(lr_fs, &lr_fs->destination, session, name, &dest_exist));
-    WT_RET_NOTFOUND_OK(
-      __live_restore_fs_has_file(lr_fs, &lr_fs->source, session, name, &source_exist));
-    WT_RET(__dest_has_stop_file(lr_fs, (char *)name, session, &have_stop));
+    WT_ASSERT(session, type == WT_FS_OPEN_FILE_TYPE_LOG || type == WT_FS_OPEN_FILE_TYPE_REGULAR);
+    __wt_verbose_debug2(session, WT_VERB_LIVE_RESTORE,
+      "Transferring %s file (%s) from source to dest.\n",
+      type == WT_FS_OPEN_FILE_TYPE_LOG ? "log" : "regular", filename);
 
```

Comment:
Why does this sit between the variable declarations?

Reply:
624c19ea6